src/System.Management.Automation/utils/Encoding.cs

```diff
@@ -0,0 +1,399 @@
+/********************************************************************++
+Copyright (c) Microsoft Corporation.  All rights reserved.
+--********************************************************************/
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Management.Automation;
+using System.Management.Automation.Provider;
+
+namespace Microsoft.PowerShell
+{
+    /// <summary>
+    /// The list of available file encodings
+    /// </summary>
+    public enum FileEncoding
+    {
+        /// <summary>
+        /// No encoding, or unset.
+        /// </summary>
+        Unknown,
```

Comment:
how would this be set back to -1, once set? would the user be stuck with the value?